src/main/java/frc/robot/sensors/Camera.java

```diff
@@ -0,0 +1,125 @@
+    }
+
+    public void getVisionOdometry(CommandSwerveDrivetrain drivetrain, Telemetry logger) {
+        List<PhotonPipelineResult> results = camera.getAllUnreadResults();
+        for (PhotonPipelineResult result : results) {
+            boolean inRange = false;
+            double highestAmbiguity = 0;
+            for (PhotonTrackedTarget target: result.getTargets()) {
+                double xSquared = target.getBestCameraToTarget().getX() * target.getBestCameraToTarget().getX();
+                double ySquared = target.getBestCameraToTarget().getY() * target.getBestCameraToTarget().getY();
+                double distance = Math.sqrt(xSquared + ySquared);
+                if(target.getPoseAmbiguity() > highestAmbiguity) {
+                    highestAmbiguity = target.getPoseAmbiguity();
+                }
+                if(distance < 4){
+                    inRange = true;
+                }
+            }
+            if(inRange && highestAmbiguity < 0.1) {
+                photonPoseEstimator.setReferencePose(drivetrain.getState().Pose);
+                Optional<EstimatedRobotPose> optionalOdoState = photonPoseEstimator.update(result);
+                if (optionalOdoState.isPresent()) {
+                    odoState = optionalOdoState.get();
+                    var newPose = odoState.estimatedPose.toPose2d();
+                    logger.updateVision(newPose);
+                }
+            }
+        }
+    }
+
+    public void updateVisionOdometry(CommandSwerveDrivetrain driveTrain, Telemetry logger){
+        getVisionOdometry(driveTrain, logger);
+        driveTrain.addVisionMeasurement(odoState.estimatedPose.toPose2d(), odoState.timestampSeconds);
+    }
+    public double getXOdoState() {
+        if(odoState != null) {
+            return odoState.estimatedPose.getX();
+        } else {
+            return 0;
+        }
+    }
+    public double getYOdoState() {
+        if(odoState != null) {
+            return odoState.estimatedPose.getY();
+        } else {
+            return 0;
+        }
+    }
+    public double getZOdoState() {
+        if(odoState != null) {
+            return odoState.estimatedPose.getZ();
+        } else {
+            return 0;
+        }
+    }
+    public double getRotationDegreesOdoState() {
+        if(odoState != null) {
+            return odoState.estimatedPose.getRotation().toRotation2d().getDegrees();
+        } else {
+            return 0;
+        }
+    }
+    public String getOdoStateQuaternion() {
+        if(odoState != null) {
+            return odoState.estimatedPose.getRotation().getQuaternion().toString();
+        } else {
+            return "null";
+        }
+    }
+
+    public void getFakeResult(CommandSwerveDrivetrain driveTrain) {
+        Pose2d fakePose = new Pose2d(2, 2, new Rotation2d(0));
+        driveTrain.addVisionMeasurement(fakePose, MathSharedStore.getTimestamp());
+    }
+
+    public void initSendable(SendableBuilder builder) {
```

Comment:
Now the Telemetry logger sends the estimated vision pose updates to SmartDashboard in more usable way. So basically none of these topics are needed anymore on the Network Tables. You could use this init sendable to show something specific for this Camera class. Something like a pose strategy chooser maybe, for trying/testing maybe? Otherwise you can remove the whole camera sendable from the SmartDashboard (in the robot container).